core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/arrayfunction/ArrayFunctionExpression.kt

```diff
@@ -0,0 +1,41 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.arrayfunction
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.ValidType
+
+sealed class ArrayFunctionExpression<T : ValidType>(
+    private val symbol: String,
+    private val array: TypeExpression<ArrayType<T>>,
+    private vararg val args: TypeExpression<out ValidType>,
+    private val extra: TypeExpression<out ValidType>? = null,
```

Comment:
this 'extra' is only used in the _ArrayReplaceExpression_ (as far as I see). Can't you remove it already there, like e.g.:
```
class ArrayReplaceExpression<T : ValidType>(
    array: TypeExpression<ArrayType<T>>,
    toReplace: TypeExpression<T>,
    replaceWith: TypeExpression<T>,
    max: TypeExpression<NumberType>? = null,
) : ArrayFunctionExpression<T>("ARRAY_REPLACE", array, *listOfNotNull(toReplace, replaceWith, max).toTypedArray())
```

Reply:
done